### PR TITLE
[SPARK-14705][YARN]support Multiple FileSystem for YARN STAGING DIR

### PR DESCRIPTION
**Pull request by lianhuiwang**
## What changes were proposed in this pull request?

In SPARK-13063, It makes the SPARK YARN STAGING DIR as configurable. But it only support default FileSystem. If there are many clusters, It can be different FileSystem for different cluster in our spark.
## How was this patch tested?

I have tested it successfully with following commands:
MASTER=yarn-client ./bin/spark-shell --conf spark.yarn.stagingDir=hdfs:namenode2/temp
$SPARK_HOME/bin/spark-submit --conf spark.yarn.stagingDir=hdfs:namenode2/temp

### CODE REVIEW

File: yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala [hunk 1]
```diff
@@ -356,8 +361,8 @@ private[spark] class Client(
     logInfo("Preparing resources for our AM container")
     // Upload Spark and the application JAR to the remote file system if necessary,
     // and add them as local resources to the application master.
-    val fs = FileSystem.get(hadoopConf)
-    val dst = getAppStagingDirPath(sparkConf, fs, appStagingDir)
+    val dst = new Path(appStagingBaseDir, appStagingDir)
```
Comment (vanzin) [inline yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala]: You could pass `appStagingDir` as a `Path` to this method and save some duplication; same for `setupLaunchEnv` below.

This whole class could use some cleanup in that regard, but these two are pretty low-hanging fruit.

**Comment by tgravescs** [comment]
test says NA, what testing have you done with this?

**Comment by lianhuiwang** [comment]
@vanzin  yes, I update code with your comments.Thanks.
@tgravescs I test this on my spark using spark-shell and spark-submit, I update it.Thanks.

**Comment by jerryshao** [comment]
So my understanding is that actually supporting **different** HDFS other than default one, not **multiple** HDFS, is that right?

**Comment by lianhuiwang** [comment]
@jerryshao Yes, what you said is right.

**Comment by HyukjinKwon** [comment]
(This is super minor but I remember I was told it might be better if those cc are added in comments not in the description because PR description is the place where to describe the PR.)

**Comment by vanzin** [comment]
@HyukjinKwon the merge scripts clean up "@" references from the PR summary.

**Comment by vanzin** [comment]
LGTM, merging to master.

**Comment by HyukjinKwon** [comment]
@vanzin Thank you!

**Comment by lianhuiwang** [comment]
@HyukjinKwon  @vanzin Thanks. I have updated PR description. But @vanzin have merged to master before. So I think it does not matter for this PR.

**Pull request closed without merging**